### PR TITLE
feat(abstract-eth): support txn bulding for v4 wallet on eth chain

### PR DESCRIPTION
**Pull request by mullapudipruthvik**
Ticket: WIN:2755

TICKET: WIN-2755

### CODE REVIEW

File: modules/sdk-coin-eth/test/resources/eth.ts [hunk 1]
```diff
@@ -94,12 +94,12 @@ export const SEND_TERC_DATA =
...
+  '0xf901cc02843b9aca0083b8a1a0948f977e912ef500548a0c3be6ddde9899f1199b8180b901643912521500000000000000000000000019645032c7f1533395d44a629462e751084d3e4c000000000000000000000000000000000000000000000000000000003b9aca0000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000005ec67e28000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000417e522e602252e010f97b81b1a83128cf283dcef2ea06c221a236c1c11883adfc222a46f4c9a6ab8cd1d5dd1e963ffa443de88c9cebf755c399702dc00b0b55d61b000000000000000000000000000000000000000000000000000000000000008284f4a0375b0731252b8a4a6f54cd3ee8078e5ba8a89168c8b9b3a49a7a3fb60984b49fa029de0bb07d891c5ad4273016055e474b0c70fa288008eadfaa16ec73b6b78427';
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-eth/test/resources/eth.ts:97]: updated the txHex to use `hteth`

File: modules/sdk-coin-eth/test/unit/transactionBuilder/send.ts [hunk 2]
```diff
@@ -100,6 +100,25 @@ describe('Eth transaction builder send', () => {
...
+        .transfer()
+        .amount(amount)
+        .to(recipient)
+        .expirationTime(expireTime)
+        .contractSequenceId(sequenceId)
+        .key(key);
+      txBuilder.sign({ key: testData.PRIVATE_KEY });
+      await txBuilder.build();
+      const operationData = txBuilder.transfer().getOperationData();
+      should.equal(operationData[1][0], '17000');
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-eth/test/unit/transactionBuilder/send.ts:119]: this tests ensure the prefix is chainId for walletVersion 4

**Review by OttoAllmendinger — DISMISSED** [review]
@BitGo/coins level changes look OK

**Review by zahin-mohammad — DISMISSED** [review]
Please remove @BitGo/wallet-platform from the CODEOWNERS for eth/evm specific modules. This has been requested in past PR's as a non-blocking change, but never addressed.

File: modules/abstract-eth/src/lib/transferBuilder.ts [hunk 3]
```diff
@@ -95,7 +101,7 @@ export class TransferBuilder {
     throw new InvalidParameterValueError('Invalid expiration time');
   }
 
-  signAndBuild(chainId?: string, coinUsesNonPackedEncodingForTxData?: boolean): string {
+  signAndBuild(chainId: string, coinUsesNonPackedEncodingForTxData?: boolean): string {
```
[the remark this answers is not in the document]
Comment (gianchandania) [reply modules/abstract-eth/src/lib/transferBuilder.ts:104]: wondering if this should be considered as a breaking change?

[the remark this answers is not in the document]
Comment (mullapudipruthvik) [reply modules/abstract-eth/src/lib/transferBuilder.ts:104]: don't we always pass chainId here
https://github.com/BitGo/BitGoJS/blob/8786ff35778da15e88e314fae9b0b9db12772dc1/modules/abstract-eth/src/lib/transactionBuilder.ts#L634

[the remark this answers is not in the document]
Comment (gianchandania) [reply modules/abstract-eth/src/lib/transferBuilder.ts:104]: Actually, there we always pass chainId, so it won't affect wallet platform and WRW most likely. But if some of our user is directly using signAndBuild method, so I was thinking it as a breaking change from that perspective

**Review by gianchandania — DISMISSED** [review]

**Review by dpkjnr — DISMISSED** [review]

**Review by the-smooth-operator — DISMISSED** [review]
approved from gitadmins PoV

**Comment by mullapudipruthvik** [comment]
rebased due to merge conflict

**Review by sumanmukherjee03 — APPROVED** [review]
codeowners change approved

File: modules/abstract-eth/src/lib/transferBuilder.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (zahin-mohammad) [reply modules/abstract-eth/src/lib/transferBuilder.ts:104]: If the API changes and is exposed publicly, it is a breaking change. Please mark the commit as such.

File: modules/abstract-eth/src/lib/transferBuilders/baseNFTTransferBuilder.ts [hunk 4]
```diff
@@ -38,6 +39,11 @@ export abstract class BaseNFTTransferBuilder {
     throw new InvalidParameterValueError('Invalid expiration time');
   }
 
+  walletVersion(version: number): this {
```
Comment (zahin-mohammad) [inline modules/abstract-eth/src/lib/transferBuilders/baseNFTTransferBuilder.ts:42]: Nit: This function scheme sounds like a getter opposed to a setter. Maybe `setWalletVersion`?

**Review by zahin-mohammad — APPROVED** [review]

**Review by gianchandania — APPROVED** [review]
lgtm

**Pull request merged**